tls/s2n_connection.c

```diff
@@ -352,6 +352,18 @@ int s2n_connection_set_config(struct s2n_connection *conn, struct s2n_config *co
         conn->multirecord_send = true;
     }
 
+    /* Users can enable OCSP status requests via s2n_config_set_status_request_type.
+     * However, s2n_config_set_verification_ca_location can also enable OCSP status
+     * requests. To ensure backwards compatibility, this function is allowed to enable
+     * OCSP status requests for clients. For servers, however, OCSP status requests
+     * are only sent if the user intentionally opted in via
+     * s2n_config_set_status_request_type.
+     */
```

Comment:
I already understand the problem, but still found this comment confusing. I think it was mostly the "this function"-- I read that as s2n_connection_set_config, not s2n_config_set_verification_ca_location. You also don't explicitly say that we can "change" the behavior for servers bc mutual auth didn't originally support ocsp stapling, which I think is a key part of the problem.

Reply:
I re-wrote this comment to try and make it better. Let me know if it's still unclear!